spiketoolkit/comparison/groundtruthstudy.py

```diff
@@ -0,0 +1,386 @@
+    
+
+    Parameters
+    ----------
+    study_folder: str
+        The study folder.
+    
+    gt_dict : a dict of tuple (recording, sorting_gt)
+        Dict of tuple that contain recording and sorting ground truth
+    """
+    
+    study_folder = Path(study_folder)
+    assert not os.path.exists(study_folder), 'study_folder already exists'
+    
+    os.makedirs(study_folder)
+    os.makedirs(study_folder / 'raw_files')
+    os.makedirs(study_folder / 'ground_truth')
+    
+    
+    for rec_name, (recording, sorting_gt) in gt_dict.items():
+        
+        # write recording as binary format + json + prb
+        raw_filename = study_folder / 'raw_files' / (rec_name+'.raw')
+        prb_filename = study_folder / 'raw_files' / (rec_name+'.prb')
+        json_filename = study_folder / 'raw_files' / (rec_name+'.json')
+        num_chan = recording.get_num_channels()
+        chunksize = 2**24// num_chan
+        sr = recording.get_sampling_frequency()
+        
+        se.write_binary_dat_format(recording, raw_filename, time_axis=0, dtype='float32', chunksize=chunksize)
+        se.save_probe_file(recording, prb_filename, format='spyking_circus')
+        with open(json_filename, 'w', encoding='utf8') as f:
+            info = dict(sample_rate=sr, num_chan=num_chan, dtype='float32', frames_first=True)
+            json.dump(info, f, indent=4)
+        
+        # write recording sorting_gt as with npz format
+        se.NpzSortingExtractor.write_sorting(sorting_gt, study_folder / 'ground_truth' / (rec_name+'.npz'))
+    
+    # make an index of recording names
+    with open(study_folder / 'names.txt', mode='w', encoding='utf8') as f:
+        for rec_name in  gt_dict:
+            f.write(rec_name + '\n')
+
+
+def get_rec_names(study_folder):
+    """
+    Get list of keys of recordings.
+    Read from the 'names.txt' file in stufy folder.
+    
+    Parameters
+    ----------
+    study_folder: str
+        The study folder.
+    
+    Returns
+    ----------
+    
+    rec_names: list
+        LIst of names.
+    """
+    with open(study_folder / 'names.txt', mode='r', encoding='utf8') as f:
+        rec_names = f.read()[:-1].split('\n')
+    return rec_names
+
+
+def get_recordings(study_folder):
+    """
+    Get ground recording as a dict.
+    
+    They are read from the 'raw_files' folder with binary format.
+    
+    Parameters
+    ----------
+    study_folder: str
+        The study folder.
+    
+    Returns
+    ----------
+    
+    recording_dict: dict
+        Dict of rexording.
+        
+    """
+    study_folder = Path(study_folder)
+    
+    rec_names = get_rec_names(study_folder)
+    recording_dict = {}
+    for rec_name in rec_names:
+        raw_filename = study_folder / 'raw_files' / (rec_name+'.raw')
+        prb_filename = study_folder / 'raw_files' / (rec_name+'.prb')
+        json_filename = study_folder / 'raw_files' / (rec_name+'.json')
```

Comment:
can we use yaml instead? :) 
Just a suggestion. I find the yaml to be more human-readable

Reply:
Yes but, yaml is not built in, no ?

Reply:
Yes you are right.. Is pyyaml in the requirements?

Reply:
I perfer to avoid a dependency for a such small thing.
This is not something we will human read often like a docker or vm setup.
This is just for dtype and num_chan.

Reply:
Ok let's stick to json then :)